client/allocrunner/taskrunner/task_runner.go

```diff
@@ -426,7 +426,11 @@ func NewTaskRunner(config *Config) (*TaskRunner, error) {
 
 	// Use the client secret only as the initial value; the identity hook will
 	// update this with a workload identity if one is available
-	tr.setNomadToken(config.ClientConfig.Node.SecretID)
+	//TODO(schmichael) can we remove this entirely? seems like a huge
+	//security problem if we use accidentally use it instead of the
+	//workload identity
```

Comment:
We probably should. But the original reason for this was because of client upgrades from pre-1.4.0. Suppose you've got a Nomad 1.3.x cluster that starts an allocation with a `template` block that reads from Nomad native service discovery. That alloc has no Workload Identity b/c that was shipped in 1.4.0 and we mint the WI on alloc creation. When you upgrade the client to 1.4.x, if we didn't fallback to the node secret then all those allocations would fail.

Reply:
k, reverting for now and we can followup in a separate pr